extractData.go

```diff
@@ -0,0 +1,37 @@
+package main
+
+import (
+	"encoding/json"
+	"log"
+	"net/http"
+	"os"
+)
+
+func extractData[T Wind | Basic](w http.ResponseWriter, filenm string, params T) error {
+	// Reading data frm the file
+	jsonData, err := os.ReadFile(filenm)
+	if err != nil {
+		log.Println("Couldn't read from file")
+		return err
+	}
+
+	// Extracting what I wanna send to the user
+	err = json.Unmarshal(jsonData, &params)
+	if err != nil {
+		log.Println("Couldn't unMarshall")
+		return err
+	}
+
+	// converting the extracted data to json again
+	dat, err := json.Marshal(params)
+	if err != nil {
+		log.Printf("Failed to marshal JSON response: %v", params)
+		return err
+	}
+
+	// Sending the Response
+	w.Header().Add("Content-Type", "application/json")
+	w.WriteHeader(200)
+	w.Write(dat)
+	return nil
```

Comment:
return data

Reply:
he is returning nil if there are no errors so that should be fine right ?

Reply:
Concern is **modularity**! I've mentioned this in the previous comment somewhere for this function.

No @adofm , the expected return value must be (whatever_data, err) because this function must merely `extractData` and return it. The fn calling `extractData()` must receive (data, nil) and do whatever it wants with it, including the act of writing to http.Response. 

In essence, extracting data and writing that data to something are two different things that require two separate routines.

Reply:
```
w.Header().Add("Content-Type", "application/json")
w.WriteHeader(200)
w.Write(dat)
``` 
Unless @Yendelevium has valid reasons to retain it, these should be removed along with the http.ResponseWriter function parameter.

Reply:
Okay

> Concern is **modularity**! I've mentioned this in the previous comment somewhere for this function.
> 
> No @adofm , the expected return value must be (whatever_data, err) because this function must merely `extractData` and return it. The fn calling `extractData()` must receive (data, nil) and do whatever it wants with it, including the act of writing to http.Response.
> 
> In essence, extracting data and writing that data to something are two different things that require two separate routines.

Okay